### DISCUSSION TITLE
How to enable timezone support for time scale?

### DISCUSSION THREAD

**Discussion by timocov** [tradingview/lightweight-charts#643] [Q&A] [answered]
Related issue #497.

Is it possible to use timezones on the chart? Like on https://www.tradingview.com/chart or https://charting-library.tradingview.com.

Related to https://stackoverflow.com/questions/65785045/how-to-set-a-custom-time-zone-for-lightweight-charts

**Comment by timocov** [comment, marked as answer]
Please read the doc https://github.com/tradingview/lightweight-charts/blob/7104e9a4fb399f18db7a2868a91b3246014c4324/docs/time-zones.md to get more information related to support time zones.

**Comment by garmoshka-mo** [comment]
[documentation about timezones][1] has a lot of sophistry (a sign that someone is sabotaging a simple task). They could easily add some property like `convertToLocalTimezone: true` or `timezoneOffset: 7200`, but it seems like they had an internal dispute and decided not to do anything. ☹️

I hope this will be fixed soon, meanwhile if you don't want to distort original data, use following formatter:

```js
const localTimezoneOffset = new Date().getTimezoneOffset() * 60

....

  timeScale: {
    timeVisible: true,
    tickMarkFormatter: (time, tickMarkType, locale) => {
      return defaultTickMarkFormatter({timestamp: time - localTimezoneOffset},
        tickMarkType, locale)
    },
  },
```

where `defaultTickMarkFormatter` is library's default formatter:
```js
function defaultTickMarkFormatter(timePoint, tickMarkType, locale) {
  const formatOptions = {};

  switch (tickMarkType) {
    case 0: //TickMarkType.Year:
      formatOptions.year = 'numeric';
      break;

    case 1: // TickMarkType.Month:
      formatOptions.month = 'short';
      break;

    case 2: //TickMarkType.DayOfMonth:
      formatOptions.day = 'numeric';
      break;

    case 3: //TickMarkType.Time:
      formatOptions.hour12 = false;
      formatOptions.hour = '2-digit';
      formatOptions.minute = '2-digit';
      break;

    case 4: //TickMarkType.TimeWithSeconds:
      formatOptions.hour12 = false;
      formatOptions.hour = '2-digit';
      formatOptions.minute = '2-digit';
      formatOptions.second = '2-digit';
      break;

    default:
      // ensureNever(tickMarkType);
  }

  const date = timePoint.businessDay === undefined
    ? new Date(timePoint.timestamp * 1000)
    : new Date(Date.UTC(timePoint.businessDay.year, timePoint.businessDay.month - 1, timePoint.businessDay.day));

  // from given date we should use only as UTC date or timestamp
  // but to format as locale date we can convert UTC date to local date
  const localDateFromUtc = new Date(
    date.getUTCFullYear(),
    date.getUTCMonth(),
    date.getUTCDate(),
    date.getUTCHours(),
    date.getUTCMinutes(),
    date.getUTCSeconds(),
    date.getUTCMilliseconds()
  );

  return localDateFromUtc.toLocaleString(locale, formatOptions);
}
```





  [1]: https://github.com/tradingview/lightweight-charts/blob/7104e9a4fb399f18db7a2868a91b3246014c4324/docs/time-zones.md

**Comment by xrado** [comment]
@garmoshka-mo solves formatting on x axis, but how do you format time shown on hover?
<img width="278" height="264" alt="Screenshot 2025-09-18 at 12 57 26" src="https://github.com/user-attachments/assets/e2c0d225-090c-4450-bfca-9011f4a1776a" />

**Reply by xrado** [reply]
setting `localization: timeFormatter:` to same function, seems to work
